lib/TAP/Harness.pm

```diff
@@ -555,8 +555,10 @@ sub runtests {
         $self->_make_callback( 'after_runtests', $aggregate );
     };
     my $run = sub {
-        $self->aggregate_tests( $aggregate, @tests );
-        $finish->();
+        my $bailout;
+        eval { $self->aggregate_tests( $aggregate, @tests ); 1 }
+            or do { $bailout = $@ || 'unknown_error' };
```

Comment:
What's the purpose of the $bailout variable, when the return value of eval will also tell you if the call succeeded or not?

Reply:
Readability. I'm not a friend of mind-wrappers such as
`$finish->( defined eval{ $self->aggregate_tests($aggregate, @tests); 1 } );`